### PR TITLE
Update serializers.py

### LINKED ISSUE #76: UID for password reset incorrectly uses base64
When a password reset link is request, the UID is encoded into the url as such, by `django allauth`
```tex
/accounts/password/reset/key/7w-1x7-d1a11z1sa1s1a1s/
```
Such that the uid is `7w` and the token is `1x7-d1a11z1sa1s1a1s`

The UID is encoded here, as **base36**:
https://github.com/pennersr/django-allauth/blob/330bf899dd77046fd0510221f3c12e69eb2bc64d/allauth/account/forms.py#L524

However, when the UID decoded in `django-rest-auth` as **base64**
https://github.com/jazzband/dj-rest-auth/blob/02c9242e3d069164692ef44a0f15eaca31a41cac/dj_rest_auth/serializers.py#L214

### PR DESCRIPTION
**Pull request by gsheni**
Fixes #76
Also uses correct token generator for checking the token

### CODE REVIEW

**Comment by iMerica** [comment]
Hi, thanks for submitting this PR. Unfortunately, this code change would result in an exception in an environment where all All Auth is not installed (because it's an **optional** dependency).

File: dj_rest_auth/serializers.py [hunk 1]
```diff
@@ -1,7 +1,6 @@
 from django.conf import settings
 from django.contrib.auth import authenticate, get_user_model
 from django.contrib.auth.forms import PasswordResetForm, SetPasswordForm
-from django.contrib.auth.tokens import default_token_generator
 from django.utils.encoding import force_text
 from django.utils.http import urlsafe_base64_decode as uid_decoder
```
Comment (mimischi) [inline dj_rest_auth/serializers.py:5]:
```suggestion
```

File: dj_rest_auth/serializers.py [hunk 2]
```diff
@@ -11,6 +10,8 @@
     from django.utils.translation import ugettext_lazy as _
 from rest_framework import exceptions, serializers
 from rest_framework.exceptions import ValidationError
+from allauth.account.forms import default_token_generator
+from allauth.account.utils import url_str_to_user_pk
```
Comment (mimischi) [inline dj_rest_auth/serializers.py:14]:
```suggestion
try:
    from allauth.account.forms import default_token_generator 
    from allauth.account.utils import url_str_to_user_pk
    decoder = url_str_to_user_pk
except ImportError:
    from django.contrib.auth.tokens import default_token_generator 
    from django.utils.http import urlsafe_base64_decode as uid_decoder
    decoder = uid_decoder
```

File: dj_rest_auth/serializers.py [hunk 3]
```diff
@@ -213,7 +214,7 @@ def validate(self, attrs):
 
         # Decode the uidb64 to uid to get User object
         try:
-            uid = force_text(uid_decoder(attrs['uid']))
+            uid = force_text(url_str_to_user_pk(attrs['uid']))
```
Comment (mimischi) [inline dj_rest_auth/serializers.py:217]:
```suggestion
            uid = force_text(decoder(attrs['uid']))
```

**Comment by gsheni** [comment]
> Just throwing out an idea: could one try to import the `django-allauth` solution and fallback to the current way if it is not installed?

I like your solution and accepted the changes.

**Comment by gsheni** [comment]
I am closing out this PR as the change no longer seem relevant.

**Comment by squio** [comment]
Question: why was this PR closed?

Looks like this solves the problem we have to use the existing allauth app for the (web based) password reset action in a really elegant way.

**Comment by gsheni** [comment]
- Closing in favor of https://github.com/Tivix/django-rest-auth/pull/643

**Pull request closed without merging**